server/src/main/java/org/elasticsearch/search/fetch/FetchSearchResult.java

```diff
@@ -28,6 +31,8 @@ public final class FetchSearchResult extends SearchPhaseResult {
 
     private ProfileResult profileResult;
 
+    private final RefCounted refCounted = LeakTracker.wrap(AbstractRefCounted.of(() -> hits = null));
```

Comment:
Obviously this doesn't do much for us GC-wise for now, but I figured if we can get anything practical out of this increment then it's probably this.